packages/flutter_tools/lib/src/base/user_messages.dart

```diff
@@ -74,8 +74,7 @@ class UserMessages {
       'Install Android Studio from: https://developer.android.com/studio/index.html\n'
       'On first launch it will assist you in installing the Android SDK components.\n'
       '(or visit ${_androidSdkInstallUrl(platform)} for detailed instructions).\n'
-      'If the Android SDK has been installed to a custom location, set $envKey to that location.\n'
-      'You may also want to add it to your PATH environment variable.\n';
+      'If the Android SDK has been installed to a custom location, please use `flutter config --android-studio-dir` or `flutter config --android-sdk` to set $envKey to that location.\n';
```

Comment:
Technically, it doesn't set an environment variable. It just reads from this config directly in https://github.com/flutter/flutter/blob/6978832293eefa7cd3c2cc064d4698f8320f80f5/packages/flutter_tools/lib/src/android/android_studio.dart#L159. I think you can just remove the `String envKey` argument too since we don't need it anymore.

Reply:
I did try removing that, but it caused some tests to fail. Specifically this test: https://github.com/flutter/flutter/blob/7f3c9b6bdab5a38cfa9eb4ba5aa5d7c5378282f3/packages/flutter_tools/test/general.shard/android/android_workflow_test.dart#L334

Reply:
Read your comment below. I'll change that test so it reflects the new error messages.